frontend/app/(footer)/dashboard/page.tsx

```diff
@@ -135,8 +173,28 @@ export default function Dashboard() {
       </div>
       <div className="flex justify-center">
         <div className="pb-10 screen-width-dashboard">
-          {eventDataList.length === 0 ? (
-            <div className="flex justify-center">
+          {loading ? (
+            <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-8 min-h-screen justify-items-center">
+              {loadingEventDataList.map((event, eventIdx) => {
```

Comment:
put a TODO comment here that this bandaid fix here of repeated code needs to be fixed after and link a new Jira ticket here :)) (ik u love jira tickets so much)